src/speculative/core.cljc

```diff
@@ -14,7 +14,7 @@
 
 (defn map-entry? [v]
   #?(:cljs (clojure.core/map-entry? v)
-     :clj #(instance? java.util.Map$Entry %)))
+     :clj (instance? java.util.Map$Entry v)))
```

Comment:
This will fail for entries from Clojure's sorted maps (and could theoretically fail for other map implementations).